src/main/java/com/example/repository/BookRepository.java

```diff
@@ -0,0 +1,10 @@
+package com.example.repository;
+
+import com.example.model.Book;
+import java.util.Optional;
+
+public interface BookRepository {
+    Book save(Book book);
+
+    Optional<Book> findAll();
```

Comment:
invalid method return type
```suggestion
    List<Book> findAll();
```